server/src/controllers/user.controllers.ts

```diff
@@ -88,18 +90,17 @@ const updateUser: RequestHandler = async (req: Request, res: Response): Promise<
 };
 
 const deleteUser: RequestHandler = async (req: Request, res: Response): Promise<void> => {
-  const { id: userId } = req.body;
+  const { id: userId } = req.params;
 
   if (userId) {
     try {
-      const data: UserMongoType | null = await User.findOneAndUpdate(
+      const data = await User.findOneAndUpdate(
```

Comment:
i think the data type here is implicitly any, can define it or I think a better approach just use updateOne. Since we don't need the delete to return anything.

can refer here https://linuxhint.com/findoneandupdate-method-mongodb/#:~:text=The%20findOneAndUpdate()%20method%20returns,and%20update%20a%20single%20document.